### PR TITLE
[WIP] Reorganize syntax and code

### PR DESCRIPTION
**Pull request by abayer**
Note - not going to merge this until after #9, so as not to make @dwnusbaum deal with reconciling everything. =)

So now we'll read from `jenkins-x.yml`, within
`PipelineConfig.Pipelines.[Release|PullRequest|Feature].Pipeline`. Tests
are updated accordingly, and I've also moved the code around -
`pkg/buildpipeline/pipeline.go` is now in
`pkg/kpipelines/syntax/pipeline.go` (tried to put it in
`pkg/kpipelines/syntax.go` but that resulted in dependency loops
between packages, so yoink, let's move to a subpackage and clear
*that* up).

Switch to using jenkins-x.yml with Pipeline embedded in it, and rework
generation from Pipeline YAML to be more like the CRDs generated from
build packs.

Note that there are still *plenty* of problems: `jx get build logs`
lists the build number but not a name, so something's messed up there,
and no `PipelineActivity` is created, so things aren't actually glued
together right yet.

### CODE REVIEW

**Review by dwnusbaum — APPROVED** [review]
Looks fine to me, although maybe `validation_failures/bad_apiVersion.yaml` can be deleted now?

**Comment by abayer** [comment]
@dwnusbaum Reworked this on top of #9, including switching `jx step create task` to not have the `--from-repo` option any more - instead, that'll just kick in if `jenkins-x.yml` exists and has a `Pipeline` defined for one of the pipeline lifecycles. Now we need tests. =)

**Review by dwnusbaum — APPROVED** [review]

**Comment by abayer** [comment]
Updated the description - I think this is as workingish as I can get it right at the moment. @dwnusbaum, thoughts?

File: pkg/kpipelines/syntax/pipeline.go [hunk 1]
```diff
@@ -4,23 +4,25 @@ import (
...
 	"k8s.io/apimachinery/pkg/api/equality"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"regexp"
-	"strings"
-	"time"
 )
 
 const (
-	PipelineFileName = "jenkins-x.yaml"
+	defaultContainerName = "maven"
```
Comment (dwnusbaum) [inline pkg/kpipelines/syntax/pipeline.go]: Just a temp thing? Not sure what would make the most sense here, but I don't know if JX has the same Java-leaning users as regular Jenkins.

**Review by dwnusbaum — APPROVED** [review]

File: pkg/kpipelines/syntax/pipeline.go (hunk 1 above)
Comment (abayer) [reply pkg/kpipelines/syntax/pipeline.go]: Yeah, I yanked that after you started reviewing. =)

**Pull request merged**